packages/lib-utils/src/ws/WSFactory.ts

```diff
@@ -0,0 +1,407 @@
+import { consoleLogger } from '@monorepo/common';
+import type {
+  EventHandlers,
+  EventHandlerTypes,
+  CloseHandler,
+  MessageHandler,
+  BulkMessageHandler,
+  MessageDataType,
+  DestroyHandler,
+  OpenHandler,
+  ErrorHandler,
+} from './types';
+import { applyConfigSubProtocols, applyConfigHost, createURL } from './ws-utils';
+
+/**
+ * The WS configurations.
+ */
+export type WSOptions = {
```

Comment:
Wondering if we should create a new file in `lib-utils/src/types/ws.ts` for the enum/types in this file? Not blocking.

Reply:
I have a types file... I could put it in there...

Reply:
> Vojtech Szocs
> I understand your point, let's do what is best and easiest. Putting types close to implementation makes sense.
> Having a separate types.ts next to specific code sounds reasonable.

This was agreed upon in slack to go the other direction and keep my local types file. Not going to do this request as a result.